src/common/binarystream.ts

```diff
@@ -127,16 +127,19 @@ export default class BinaryStream {
 
     const uint8array = this.uint8array;
     const index = this.index;
-    let data = '';
+    let chars = [];
 
     for (let i = 0; i < bytes; i++) {
-      data += String.fromCharCode(uint8array[index + i]);
+      chars.push(uint8array[index + i])
     }
 
     this.index += bytes;
     this.remaining -= bytes;
 
-    return data;
+    return chars.map((ch) => {
+      if (ch === 0x0) return '0';
+      return String.fromCharCode(ch);
+    }).join('');
```

Comment:
Why are we iterating over the entire stream twice? Couldn't this logic be moved over to the loop?

Reply:
Removed one loop